spec/integration/data_stream/manifest.spec.yml

```diff
@@ -115,39 +115,51 @@ spec:
           default:
             description: Default value(s) for variable
             $ref: "#/definitions/input_variable_value"
-        if:
-          properties:
-            type:
-              const: select
-        then:
-          required:
-            - options
-          properties:
-            options:
-              description: List of options for select type
-              type: array
-              items:
-                type: object
-                additionalProperties: false
-                properties:
-                  value:
-                    type: string
-                    examples:
-                    - node
-                    - cluster
-                  text:
-                    type: string
-                    examples:
-                    - node
-                    - cluster
-                required:
-                - value
-                - text
-              min_items: 1
-        else:
-          not:
-            required:
-              - options
+        allOf:
+         - if:
+             properties:
+               type:
+                 const: select
+           then:
+             required:
+               - options
+             properties:
+               options:
+                 description: List of options for select type
+                 type: array
+                 items:
+                   type: object
+                   additionalProperties: false
+                   properties:
+                     value:
+                       type: string
+                       examples:
+                       - node
+                       - cluster
+                     text:
+                       type: string
+                       examples:
+                       - node
+                       - cluster
+                   required:
+                   - value
+                   - text
+                 min_items: 1
+           else:
+             not:
+               required:
+                 - options
+         - if:
+             anyOf:
+               - properties:
+                   name:
+                     pattern: "(password|api_key|access_key|token)"
```

Comment:
@kpollich @nimarezainia this is the regexp pattern for variable names that we would detect as candidates to be secrets, is there any other string do you think we should add here?
Should we match with exact values instead of with substrings?

Reply:
We could add the substring `secret` here as it will catch things like an AWS `secret_access_key` value. I think using substring matching is good to keep this generic.

Reply:
Added `secret` and `passphrase`, the latter is sometimes used for encrypted TLS keys.

Reply:
We should juste be cautious with values like "secret_path" which might be caught by this check though.

Reply:
> We should juste be cautious with values like "secret_path" which might be caught by this check though.

Yep, in these cases they will need to set `secret: false`.